pkg/daemon/update.go

```diff
@@ -432,7 +432,8 @@ func (dn *Daemon) updateKernelArguments(oldConfig, newConfig *mcfgv1.MachineConf
 		return nil
 	}
 	if dn.OperatingSystem != machineConfigDaemonOSRHCOS {
-		return fmt.Errorf("Updating kargs on non-RHCOS nodes is not supported: %v", diff)
+		glog.V(2).Info("Updating kargs of non RHCOS nodes is not supported")
```

Comment:
Hm.  It feels more to me like erroring out is the correct thing.  Probably?
I dunno.  This MCD-on-not-RHCOS is a super messy topic...

Is there a specific reason you're making this change?  I just want to understand this more.

The reason we silently ignore `osImageURL` on not-RHCOS is because it's always there.  But at least today kargs are not always there.

When the installer changes to use `KernelArguments` for `nosmt`, is it really right for us to silently ignore that request for BYO RHEL?

Reply:
oh well, the only reasoning was to emulate osimageurl but your point that Kargs aren't always there makes total sense, dropping that commit! Thanks!

Reply:
done

Reply:
I keep vacillating on this; following the installer-nosmt scenario above though, since the MCD today doesn't handle kargs for not-RHCOS, the user would need to use Ansible or whatever for that, but the MCD would go into a degraded state and not handle the parts it can handle (files).

So...maybe your change is right?  I dunno.

I guess perhaps the more ideal thing would be for us to *inspect* the host and not make a change if we didn't need to - then the MCD would be degraded until Ansible ran.

Reply:
> I guess perhaps the more ideal thing would be for us to _inspect_ the host and not make a change if we didn't need to - then the MCD would be degraded until Ansible ran.

I'm leaning towards this indeed. This way an admin can see something is wrong in the pool and take action instead of silently wtf'ing because kargs isn't applied (same for fips at this point)

Reply:
Yeah, it seems clearly better, but would require reworking our kargs code.  Anyways this PR as is looks clearly right.

Reply:
`I guess perhaps the more ideal thing would be for us to inspect the host and not make a change if we didn't need to `

just to understand, inspect the host to see if we need to make the kargs changes? or...?